templates/FFV_form_factors.cpp.in

```diff
@@ -424,4 +616,364 @@ std::valarray<std::complex<double>> FFV_FFS<Fj, Fi, V, FA, FB, S>::value(
+
+   // loop over all possible particle generations attached to both vertices
+   std::valarray<std::complex<double>> res {0.0, 0.0, 0.0, 0.0};
+   for (const auto& indexIn: index_range<VertexFBarFjGBar>()) {
+      for (const auto& indexOut: index_range<VertexFiBarFV>()) {
+          
+         // cycle if generations of external fermions are different then requested
+         const auto jFieldIndices = VertexFBarFjGBar::template indices_of_field<2>(indexIn);
+         const auto iFieldIndices = VertexFiBarFV::template indices_of_field<0>(indexOut);
+         if (jFieldIndices != indices_in || iFieldIndices != indices_out)
+            continue;
+         
+         // match indices of the fermion in the loop
+         const auto fermionFieldIndicesIn = VertexFBarFjGBar::template indices_of_field<0>(indexIn);
+         const auto fermionFieldIndicesOut = VertexFiBarFV::template indices_of_field<2>(indexOut);
+         if (fermionFieldIndicesIn != fermionFieldIndicesOut)
+            continue;
+
+         // cannot match indices if scalar is not would-be-goldstone corresponding to vector
+         const auto goldstoneFieldIndicesIn = VertexFBarFjGBar::template indices_of_field<1>(indexIn);
+         const auto vectorFieldIndicesOut = VertexFiBarFV::template indices_of_field<1>(indexOut);
+         // if (goldstoneFieldIndicesIn != vectorFieldIndicesOut)
+         //    continue;
+
+         // ignore contribution if both vector and fermion are SM particles
+         if (discard_SM_contributions) {
+            if (isSMField<GA>(goldstoneFieldIndicesIn) && isSMField<VB>(vectorFieldIndicesOut) && isSMField<F>(fermionFieldIndicesIn)) {
+               continue;
+            }
+         }
+   
+         const auto vertexIn = VertexFBarFjGBar::evaluate(indexIn, context);
+         const auto vertexOut = VertexFiBarFV::evaluate(indexOut, context);
+
+         const auto indexEmit = concatenate(goldstoneFieldIndicesIn, vectorFieldIndicesOut);
+         const auto vertexEmit = VertexVBarGPBar::evaluate(indexEmit, context);
+   
+         const auto mV = context.mass<GA>(goldstoneFieldIndicesIn);
+         const auto mG = context.mass<VB>(vectorFieldIndicesOut);
+         const auto mF = context.mass<F>(fermionFieldIndicesIn);
+         const auto x = Power2(mF/mV);
+         const auto y = Power2(mF/mG);
+
+         // scalar-vector-vector vertex
+         // check that definition and sign is correct
+         std::complex<double> vector_boson_coupling {-vertexEmit.value()};
+
+         const std::complex<double> A1L = 1./mj * vertexIn.right() * vertexOut.left() * OneLoopFunctionN(x,y) / 2.;
+         const std::complex<double> A2L = 1./mj * vertexIn.left() * vertexOut.right() * OneLoopFunctionN(x,y) / 2.;
+      
+         const std::complex<double> A1R = 1./mj * vertexIn.left() * vertexOut.right() * OneLoopFunctionN(x,y) / 2.;
+         const std::complex<double> A2R = 1./mj * vertexIn.right() * vertexOut.left() * OneLoopFunctionN(x,y) / 2.;
```

Comment:
@dhjjacob, when you're doing an expansion in `mj` and `mi`, are you keeping both of them to linear order? I'm asking because this is something I messed up at first. We keep masses `mj` and `mi` separate, because the same routines are also used for `\mu \to e \gamma`. There you don't need terms proportional to `mi`. But for g-2, `mi = mj`, so we have to expand to order 1 both in `mj` and `mi`.

Reply:
@Wojtek, thanks for checking.  The contributions here come from expanding mi, mj, and the loop momentum squared (q.q) about 0 to first order.